autoblocks/_impl/global_state.py

```diff
@@ -0,0 +1,70 @@
+import asyncio
+import threading
+from typing import Coroutine
+from typing import List
+from typing import Optional
+
+import httpx
+
+_client: Optional[httpx.AsyncClient] = None
+_loop: Optional[asyncio.AbstractEventLoop] = None
+_started: bool = False
+
+
+def init(client: httpx.AsyncClient) -> None:
+    global _client, _loop, _started
+
+    if _started:
+        return
+
+    _client = client
+
+    _loop = asyncio.new_event_loop()
+
+    background_thread = threading.Thread(
+        target=run_event_loop,
+        args=(_loop,),
+        daemon=True,
+    )
+    background_thread.start()
+
+    _started = True
+
+
+def run_event_loop(_loop: asyncio.AbstractEventLoop) -> None:
```

Comment:
i'd rename the `_loop` arg to something else so that it doesn't have the same name as the global